### PR TITLE
automl: add samples for language and vision

### PR DESCRIPTION
**Pull request by davidcavazos**
Adding samples for AutoML:
* Language entity extraction
* Language sentiment analysis
* Language text classification
* Translate
* Vision classification
* Vision object detection

Tests will be coming soon in a follow up PR

### CODE REVIEW

File: automl/README.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  * [Edge](https://cloud.google.com/vision/automl/docs/edge-quickstart)
+  * [Object Detection](https://cloud.google.com/vision/automl/object-detection/docs)
+* [AutoML Tables](https://cloud.google.com/automl-tables/docs)
+
+This API is part of the larger collection of Cloud Machine Learning APIs.
+
+These Go samples demonstrates how to access the Cloud AutoML API using the
+[Google Cloud Client Library for Go](https://github.com/googleapis/google-cloud-go).
+
+## Sample Types
```
Comment (tbpg) [inline automl/README.md]: For a future PR, so we can get this one in:

Do we need anything after this line? It's very easy to forget to update READMEs and we've generally avoided putting anything that requires updating in them. Pointers to the official docs where these samples are embedded is OK.

File: automl/language_batch_predict.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	automlpb "google.golang.org/genproto/googleapis/cloud/automl/v1beta1"
+)
+
+// languageBatchPredict does a batch prediction.
+func languageBatchPredict(w io.Writer, projectID string, location string, modelID string, inputURI string, outputURI string) error {
+	// projectID := "my-project-id"
+	// location := "us-central1"
+	// modelID := "TRL123456789..."
+	// input_uri := "gs://BUCKET_ID/path_to_your_input_file.txt"
+	// output_uri := "gs://BUCKET_ID/path_to_save_results/"
```
Comment (tbpg) [inline automl/language_batch_predict.go]: `inputURI` and `outputURI`

File: automl/vision_batch_predict.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	automlpb "google.golang.org/genproto/googleapis/cloud/automl/v1beta1"
+)
+
+// visionBatchPredict does a batch prediction.
+func visionBatchPredict(w io.Writer, projectID string, location string, modelID string, inputURI string, outputURI string) error {
+	// projectID := "my-project-id"
+	// location := "us-central1"
+	// modelID := "ICN123456789..."
+	// input_uri := "gs://BUCKET_ID/path_to_your_input_file.txt"
+	// output_uri := "gs://BUCKET_ID/path_to_save_results/"
```
Comment (tbpg) [inline automl/vision_batch_predict.go]: `inputURI` and `outputURI`

File: automl/vision_object_detection_predict.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	automl "cloud.google.com/go/automl/apiv1beta1"
+	automlpb "google.golang.org/genproto/googleapis/cloud/automl/v1beta1"
+)
+
+// visionObjectDetectionPredict does a prediction for image classification.
+func visionObjectDetectionPredict(w io.Writer, projectID string, location string, modelID string, filePath string) error {
+	// projectID := "my-project-id"
+	// location := "us-central1"
+	// modelID := "IOD123456789..."
+	// file_path := "testdata/salad.jpg"
```
Comment (tbpg) [inline automl/vision_object_detection_predict.go]: Do we need to check in the binary files? I'd prefer to leave them out and download them as-needed form a bucket for tests.

File: automl/language_batch_predict.go (hunk 2 above)
Comment (davidcavazos) [reply automl/language_batch_predict.go]: Copy-paste error, thanks!

File: automl/vision_batch_predict.go (hunk 3 above)
Comment (davidcavazos) [reply automl/vision_batch_predict.go]: Fixed, thanks!

File: automl/vision_object_detection_predict.go (hunk 4 above)
Comment (davidcavazos) [reply automl/vision_object_detection_predict.go]: This was going to be used for tests anyways so I'll remove them for now

File: automl/README.md (hunk 1 above)
Comment (davidcavazos) [reply automl/README.md]: I was following the Python canonicals, the AutoML samples aren't as straightforward as other samples so I think all languages are including a similar README to explain how they are organized.

**Comment by davidcavazos** [comment]
Reorganized file names and fixed some wrong input variable names

File: automl/README.md (hunk 1 above)
Comment (tbpg) [reply automl/README.md]: With the updated filenames, this is now out of date. :/ Already merged, so will you send a followup? I still think we can delete it.

Comment (davidcavazos) [reply automl/README.md]: Fixing in #1087

**Pull request merged**